contrib/labstack/echo/echotrace_test.go

```diff
@@ -0,0 +1,126 @@
+package echo
+
+import (
+	"errors"
+	"net/http/httptest"
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/labstack/echo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestChildSpan(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	router := echo.New()
+	router.Use(Middleware("foobar"))
+	router.GET("/user/:id", func(c echo.Context) error {
+		_, ok := tracer.SpanFromContext(c.Request().Context())
+		assert.True(ok)
+		return nil
+	})
+
+	r := httptest.NewRequest("GET", "/user/123", nil)
+	w := httptest.NewRecorder()
+
+	router.ServeHTTP(w, r)
+}
+
+func TestTrace200(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	router := echo.New()
+	router.Use(Middleware("foobar"))
+	router.GET("/user/:id", func(c echo.Context) error {
+		// assert we patch the span on the request context.
+		span, ok := tracer.SpanFromContext(c.Request().Context())
+		assert.True(ok)
+		span.SetTag("test.echo", "echony")
+		assert.Equal(span.(mocktracer.Span).Tag(ext.ServiceName), "foobar")
+		id := c.Param("id")
+		return c.String(200, id)
+	})
+
+	r := httptest.NewRequest("GET", "/user/123", nil)
+	w := httptest.NewRecorder()
+
+	// do and verify the request
+	router.ServeHTTP(w, r)
+	response := w.Result()
+	assert.Equal(response.StatusCode, 200)
```

Comment:
What is the purpose of this assertion? Are we testing that Echo works? We shouldn't, right? That's for them to test.